python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,262 @@
+#
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=E0401,protected-access
+
+import logging
+import numpy as np
+
+from pyspark.ml import Estimator
+from pyspark.ml.param import Param, Params, TypeConverters
+
+from sparkdl.image.imageIO import imageStructToArray
+from sparkdl.param import (
+    keyword_only, HasInputCol, HasOutputCol, SparkDLTypeConverters, 
+    CanLoadImage, HasKerasModel, HasTensorOutput)
+from sparkdl.transformers.keras_image import KerasImageFileTransformer
+import sparkdl.utils.jvmapi as JVMAPI
+from sparkdl.utils.keras_model import *
+
+logger = logging.getLogger('sparkdl')
+
+
+class KerasImageFileEstimator(Estimator, HasInputCol, HasOutputCol,
+                              CanLoadImage, HasKerasModel, HasTensorOutput):
+    
+    labelCol = Param(Params._dummy(), "labelCol",
+                     "Name of the column storing the training data labels",
+                     typeConverter=TypeConverters.toString)
+
+    optimizer = Param(Params._dummy(), "optimizer",
+                      "named Keras optimizer (str)", typeConverter=TypeConverters.toString)
+
+    loss = Param(Params._dummy(), "loss",
+                 "named Keras loss (str)", typeConverter=TypeConverters.toString)
+
+    @keyword_only
+    def __init__(self, inputCol=None, outputCol=None, labelCol=None, modelFile=None,
+                 imageLoader=None, outputMode="vector", optimizer=None, loss=None,
+                 kerasParams=None):
+        """
+        __init__(self, inputCol=None, outputCol=None, labelCol=None, modelFile=None,
+                 imageLoader=None, outputMode="vector", optimizer=None, loss=None,
+                 kerasParams=None):
+        """
+        super(KerasImageFileEstimator, self).__init__()
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+
+    @keyword_only
+    def setParams(self, inputCol=None, outputCol=None, labelCol=None, modelFile=None,
+                  imageLoader=None, outputMode="vector", optimizer=None, loss=None,
+                  kerasParams=None):
+        """
+        setParams(self, inputCol=None, outputCol=None, labelCol=None, modelFile=None,
+                  imageLoader=None, outputMode="vector", optimizer=None, loss=None,
+                  kerasParams=None):
+        """
+        kwargs = self._input_kwargs
+        self._set(**kwargs)
+        return self
+
+    def setLabelCol(self, value):
```

Comment:
for usability, we should have a `setModel` that takes a Keras model and does all the file writing. This is how most users will want to interact with it. Can you add a TODO for the future?